index.html

```diff
@@ -1,414 +1,482 @@
-				<p>
-					The approach is to add a <code>-d-dir</code> to the [[bcp47]] language tag. The language tag would look like <code>lang-d-dir</code>, where <code>lang</code> is a valid [[bcp47]] language tag, and <code>dir</code> pattern MUST have the value of <code>ltr</code>, <code>rtl</code>, or <code>auto</code>, with the semantics as defined for the <a data-cite=!html/dom.html#the-dir-attribute><code>dir</code> attribute</a> in&nbsp;[[html]].
-				</p>
-			</section>
+                <p>in Turtle, and</p>
 
-			<section>
-				<h2>Serializations</h2>
+                <pre class=nohighlight>
+						"ex:example" : {
+						    "@value" : "<bdo dir="ltr">&#x05E4;&#x05E2;&#x05D9;&#x05DC;&#x05D5;&#x05EA; &#x05D4;&#x05D1;&#x05D9;&#x05E0;&#x05D0;&#x05D5;&#x05DD;, W3C</bdo>",
+						    "@language" : "he-x-d-rtl"
+						}
+					</pre>
 
-				<p>
-					In terms of serializations, this change would be completely transparent for RDF. Ie, the example used before is:
-				</p>
+                <p>in JSON-LD.</p>
+            </section>
+
+            <section>
+                <h2>Pros and Cons</h2>
+
+                <p>
+                    A major advantage of this approach are similar to <a href=#extending-bcp></a>: it is fully transparent to current RDF deployments, and it affects only application layers that do display the literal values for humans and it also works for parsers “out of the box. Because it is explicitly restricted to some specifications, it can (and should) be ignored by HTML, CSS, or other usage areas of [[bcp47]].
+                </p>
+
+                <p>
+                    Just as for <a href=#extending-bcp></a>, a major disadvantage is that [[bcp47]] is already a very complex specification, in terms of the rich "metadata" that it assigns to languages. It is also not clear whether it "fits" the purpose of&nbsp;[[bcp47]], which is to describe metadata about language, scripts used, etc., and not a feature that is relevant for the <em>display</em> of that string. Furthermore, although it is restricted to a specific family of specification, its usage may “spill over” to other areas, creating confusion.
+                </p>
+                <p>
+                    Finally, just as for <a href=#extending-bcp></a>, the interplay between current data deployment and the assignment of such extended language tags may not be obvious.
+                </p>
+            </section>
+        </section>
+        <section id=script-subtag>
+            <h2>Rely on current BCP47 only</h2>
+
+            <p>
+				A different approach stems from the observation that the current BCP-47 can be used, unchanged, to express the necessary base direction using, if necessary, the “script” subtag. For example, the tag <code>he-Hebr</code> reinforces, through the presence of the <code>Hebr</code> subtag, that the language is not only Hebrew, but the string uses the Hebrew script, i.e., it <em>is</em> right-to-left. In other words:
```

Comment:
A better example here would be `az-Arab`, since `he` on its own should be sufficient for Hebrew, and it's best not to encourage `he-Hebr` usage.

Reply:
But... isn't the point of "Rely on BCP-47" the fact that a, possibly redundant, information can be used as an extra info reinforcing the rtl nature? That is what I try to say in:

> For example, the tag he-Hebr reinforces, through the presence of the Hebr subtag, that the language is not only Hebrew, but the string uses the Hebrew script, i.e., it is right-to-left

If what you say is true, i.e., the "he" tag is enough to remove all ambiguities of the Bidi problem (ie, the fact that the string may with a Latin character) then I still do not understand why we need the directional information in the first place... Are all our examples wrong (provided the language tag is used)?

Reply:
To think a bit further: _if_ I understand what you seem to suggest in your comment, isn't it correct to say that _the base direction information is only necessary in the absence of a correct language tag_ (by 'correct' I mean including the Script tag in case of ambiguities like Azeri). If that is indeed the case then I really do not see why this information is really necessary for applications like JSON/RDF, because that extra tag carries, in 90% of the cases, redundant information...

Reply:
> To think a bit further: if I understand what you seem to suggest in your comment, isn't it correct to say that the base direction information is only necessary in the absence of a correct language tag (by 'correct' I mean including the Script tag in case of ambiguities like Azeri). If that is indeed the case then I really do not see why this information is really necessary for applications like JSON/RDF, because that extra tag carries, in 90% of the cases, redundant information...

That's what I was thinking... and to be honest, it seems like the problems we've been discussing have arisen *because* a feature has been added to express direction separately from language in some syntaxes. If that had never happened, there would be no reason to worry about what happens when a direction is set (or set "incorrectly") without a language. It seems that the answer is to just specify the proper language if you want your data to be understood correctly.

In other words, if it makes no sense to specify a direction that contradicts the language then the two never should have been decoupled. If there are "non-language" strings that need a special direction to understand them, then, again, no special "direction" feature should have been created for this, rather, it should have used a data type (or something equivalent). The hack, at least to me, seems to be the notion of "direction" as its own property.

Reply:
@dlongley, I think the problem is a bit more nuanced than that...

Consider the acronym "W3C". Making it clear that the direction is ltr is important. However, there is no natural language that MUST be assigned to this acronym: indeed, both "W3C"@en and "W3C"@fr are acceptable and the difference between the two is important because it may decide how a text-to-speech engine would utter the acronym. Ie, direction and language are not 100% bound. Even "W3C"@ar should be acceptable for the very same reason but, in this case, we _do_ need to give a direction somehow. The question is whether "W3C'@ar-Latn is acceptable for this case or not. If it _is_, then it signals the left-to-right typesetting following the proposal in this section.